src/schemaToPropTypes.js

```diff
@@ -0,0 +1,195 @@
+let INDENT_LEVEL = 1;
+const COMPONENT_NAME_SUFFIX = 'PropTypes';
+const INDENT_CHAR = '\t';
+const QUOTE_CHAR = "'";
+
+/**
+ * Formats the component name based on React standards and a suffix to avoid name collisions.
+ * @param {String} name - The key string from the schema.
+ * @returns {string}
+ */
+const formatComponentName = name =>
+	`${name.charAt(0).toUpperCase()}${name.slice(1)}${COMPONENT_NAME_SUFFIX}`;
+
+/**
+ * Generates the required amount of indentation.
+ * @param {Number} indentLevel - The amount of indentation.
+ * @returns {string} - A string with indents
+ */
+const getIndentation = (indentLevel = INDENT_LEVEL) =>
+	indentLevel && typeof indentLevel === 'number'
+		? [...Array(Math.trunc(indentLevel))].map(() => `${INDENT_CHAR}`).join('')
+		: '';
+
+/**
+ * Decorator to add indentation to the strings output from any needed functions.
+ * @param {Function} func - The function returning a string whose return needs to be decorated.
+ * @param {Boolean} prefix - Whether the indentation should happen as a prefix or as a suffix. Prefix is default
+ * @returns {function(args: Array): string} - The decorated String
+ * @private
+ */
+const _indentDecorator = (func, prefix = true) => args => {
+	const indents = getIndentation();
+
+	return `${prefix ? indents : ''}${func.apply(this, args)}${prefix ? '' : indents}`;
+};
+
+/**
+ * Adds the `Required` string given a property and an array of required properties.
+ * @param {String} propName - The name of the property to be checked.
+ * @param {Object} property - The property itself (is property.required a valid openAPI format?).
+ * @param {Array} requiredProps - The array of required properties in a schema.
+ * @returns {string}
+ */
+const getRequired = (propName, property, requiredProps = []) =>
+	Array.isArray(requiredProps) && (requiredProps.indexOf(propName) > -1 || property.required)
+		? '.isRequired,'
+		: ',';
+
+/**
+ * Returns the name of the reference.
+ * @param {String} ref - The `$ref` string.
+ * @returns {string} - The name of the reference in the `$ref`
+ */
+const getRef = ref => ref.split('/').pop();
+
+/**
+ * Creates the string for the value of a property.
+ * @param {String} propertyName - The key (name) of the property.
+ * @param {Object} property - The property to generate the value from.
+ * @returns {string}
+ */
+const getPropTypeValue = (propertyName, property) => {
+	let str = 'PropTypes.';
+
+	switch (property.type) {
+		case 'array':
+			if (property.items.$ref) {
+				const extractRefProp = formatComponentName(getRef(property.items.$ref));
+				str += `arrayOf(${extractRefProp})`;
```

Comment:
I think here if we have a `ref` we need:
```suggestion
				str += `arrayOf(${extractRefProp})`;
				str += `arrayOf(PropTypes.shape(${extractRefProp}))`;
```